firebase-firestore/src/main/java/com/google/firebase/firestore/SnapshotListenOptions.java

```diff
@@ -0,0 +1,195 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore;
+
+import static com.google.firebase.firestore.util.Preconditions.checkNotNull;
+
+import android.app.Activity;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.firebase.firestore.util.Executors;
+import java.util.concurrent.Executor;
+
+/**
+ * An options object that configures the behavior of {@code addSnapshotListener()} calls. Instances
+ * of this class control settings such as whether metadata-only changes trigger events, the
+ * preferred data source (server or cache), and the executor for listener callbacks.
+ */
+public final class SnapshotListenOptions {
+  /** Indicates whether metadata-only changes should trigger snapshot events. */
+  private final MetadataChanges metadataChanges;
+  /** Indicates the source that we listen to. */
+  private final ListenSource source;
+  /** The executor to use to call the listener. */
+  private final Executor executor;
+  /** The activity to scope the listener to. */
+  private final Activity activity;
+
+  private SnapshotListenOptions(Builder builder) {
+    this.metadataChanges = builder.metadataChanges;
+    this.source = builder.source;
+    this.executor = builder.executor;
+    this.activity = builder.activity;
+  }
+
+  /**
+   * Returns the setting for whether metadata-only changes should trigger snapshot events.
+   *
+   * @return The setting for whether metadata-only changes should trigger snapshot events.
+   */
+  @NonNull
+  public MetadataChanges getMetadataChanges() {
+    return metadataChanges;
+  }
+
+  /**
+   * Returns the preferred source for retrieving data in snapshot listeners.
```

Comment:
Capitalize "The..."

Reply:
you mean remove "Returns" and capitalize "the"?